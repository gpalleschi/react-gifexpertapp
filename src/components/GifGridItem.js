import React from 'react'
import PropTypes from 'prop-types'; // IMPORT


export const GifGridItem = ( {url, title} ) => {
  return (
    <div className="card animate__animated animate__fadein">
         <img src={url} alt={title}></img>
	 <p>{ title }</p>
    </div>
  )
}

// SOTTO IL PROP TYPES :
GifGridItem.propTypes = {

url : PropTypes.string.isRequired,
title : PropTypes.string.isRequired
}

export default GifGridItem;



