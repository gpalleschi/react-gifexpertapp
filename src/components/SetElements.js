import React from 'react';

import PropTypes from 'prop-types';

const SetElements = ({setElements}) => {

	const handleChange = (e) => {
		setElements(e.target.value)
	}

	return (<div className='elementSelection'>
	        <label for="elements">Gif Elements</label>
		<select ClassName='eComboBox' onChange={handleChange} name="elements" id="elements">
                        <option value="1">1</option>
                        <option value="5">5</option>
                        <option value="10" selected="selected">10</option>
                        <option value="15">15</option>
                        <option value="20">20</option>
                        <option value="30">30</option>
                        <option value="40">40</option>
                        <option value="50">50</option>
                        <option value="100">100</option>
                </select>
	        </div>
		);
}

SetElements.prototype = {
	setElements: PropTypes.func.isRequired
}

export default SetElements;