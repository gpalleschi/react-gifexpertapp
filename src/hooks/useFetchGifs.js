import { useState, useEffect } from 'react';
import { getGifs } from '../helpers/getGifs';

export const useFetchGifs = ( category, elements ) => {

	const [state, setState] = useState({
		data: [],
		loading: true
	});

	useEffect(() => {
		getGifs(category, elements).then( imgs => setState( {
			data: imgs,
			loading: false
		}))
        }, [category]);

	return state; // {data, loading}
}