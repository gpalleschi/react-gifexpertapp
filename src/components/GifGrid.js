import React from 'react'
import { useFetchGifs } from '../hooks/useFetchGifs'
//import { getGifs } from '../helpers/getGifs';
import { GifGridItem } from './GifGridItem';
import PropTypes from 'prop-types';

export const GifGrid = ( {category, elements} ) => {

  const { data:images, loading } = useFetchGifs( category, elements );

  return (
    <>
    <h3>{category}</h3>

    { loading && <p>Loading ...</p>}

    { 
    <div className="card-grid"> 
	 { images.map( img => <GifGridItem key={ img.id } { ...img }/> )}
    </div>
    }
    </>	  
  )
}


GifGrid.propTypes = {
  category : PropTypes.string.isRequired,
  elements : PropTypes.number.isRequired
}

export default GifGrid;