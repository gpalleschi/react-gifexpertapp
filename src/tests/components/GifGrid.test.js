import React from 'react';
import { shallow } from 'enzyme';
import { GifGrid } from '../../components/GifGrid';
import '@testing-library/jest-dom';
import { useFetchGifs } from '../../hooks/useFetchGifs';
jest.mock('../../hooks/useFetchGifs');

describe('Prueba <GifGrid />', () => { 

	const category = 'One Punch';
	const elements = 1;

	test('Snapshot', () => { 


	    useFetchGifs.mockReturnValue({
		data: [],
		loading: true
	    });

	    const wrapper = shallow( <GifGrid category={category} elements={elements}/>);
	    expect(wrapper).toMatchSnapshot();
	})


	test('debe de mostrar items cuando se cargan imagenes useFetchGifs()', () => {

	    const gifs = [{
		    id: 'ABC',
		    url: 'https:\\qualsiasicosa1.com',
		    title: 'qualsiasi cosa 1'
	    },
	    {
		    id: '123',
		    url: 'https:\\qualsiasicosa2.com',
		    title: 'qualsiasi cosa 2'
	    }
	]

	    useFetchGifs.mockReturnValue({
		data: gifs,
		loading: false
	    });
	    
	    const wrapper = shallow( <GifGrid category={category} elements={elements}/>);

	    expect( wrapper).toMatchSnapshot();
	    // Quando è caricata la lista di gifs il parragrafo loading non deve essere presente
	    expect( wrapper.find('p').exists()).toBe(false);
	    // Se voglio validare se è presente un componente semplicemente lo scrivo
	    expect( wrapper.find('GifGridItem').length).toBe(gifs.length);


	 })
})