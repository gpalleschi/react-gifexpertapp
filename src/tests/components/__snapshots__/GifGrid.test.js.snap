// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prueba <GifGrid /> Snapshot 1`] = `
<Fragment>
  <h3>
    One Punch
  </h3>
  <p>
    Loading ...
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;

exports[`Prueba <GifGrid /> debe de mostrar items cuando se cargan imagenes useFetchGifs() 1`] = `
<Fragment>
  <h3>
    One Punch
  </h3>
  <div
    className="card-grid"
  >
    <GifGridItem
      id="ABC"
      key="ABC"
      title="qualsiasi cosa 1"
      url="https:\\\\qualsiasicosa1.com"
    />
    <GifGridItem
      id="123"
      key="123"
      title="qualsiasi cosa 2"
      url="https:\\\\qualsiasicosa2.com"
    />
  </div>
</Fragment>
`;
